### ISSUE TITLE
Model is not learning and epoch loss high

### ISSUE THREAD

**Issue by jpainam** [kaize0409/Meta-GDN_AnomalyDetection#1]
Hi, I'm trying to reproduce your work. I used the same environment as described in the readme file. But After every iteration, the epoch loss is always 80.0000 and the running i-th epoch always start very high (e.g. 2387.71
I think the model isn't learning anything. I'm using graphs/yelp dataset.
Is there something i'm doing wrong?
Thank you.
Here is the output
```
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
epoch loss 80.000000
AUC-ROC: 0.5237, AUC-PR: 0.0522, AP: 0.0531
G-dev auc_roc: 0.52374, auc_pr: 0.05308
maml training...
In 4-th run...
Total #trainable tensors:  5121025
Running 1-th epoch
Epoch loss: 2387.719971
Running 2-th epoch
Epoch loss: 2311.382324
Running 3-th epoch
Epoch loss: 2248.589844
Running 4-th epoch
Epoch loss: 2179.625488
Running 5-th epoch
Epoch loss: 2118.198730
Running 6-th epoch
Epoch loss: 2054.575195
Running 7-th epoch
```

**Comment by BlueWhaleZhou** [comment]
This issue could be a problem of feature normalization. You may want to try normalizing the node attribute matrix and/or different weight initialization.

**Comment by jpainam** [comment]
Hi. I can normalize the attribute features in the `graph_construction.py` while injecting the anomalies as follows
```python
data["Network"] = adj_s
feature_s = F.normalize(torch.from_numpy(feature_s))
data["Attributes"] = feature_s.numpy()
data["Label"] = label_s
...
```
But during the loading of data, you also normalize the feature here.
https://github.com/kaize0409/Meta-GDN_AnomalyDetection/blob/575d01b4bc76be315ededb174bc66bc8b2422c93/process_task.py#L56

So, I'm wondering if this is not double normalization. I didn't see any improvement when using `F.normalize(...)`
Any suggestion where to do the note attribute normalization is welcomed.

Thank you.

**Comment by jpainam** [comment]
Hi. Can you provide your training logs? Just to give some details about the training convergence.
In my first comment. The training loss from the `SGC` model computed by `deviation_loss` is always constant (i.e. 80.000) no matter the dataset I used.
Any advice?

**Comment by BlueWhaleZhou** [comment]
I believe it is an issue with the weights initialization/normalization in leaner.py. Since we manually initialize the weights of the model instead of using torch.nn modules, we need to re-normalize them after initialization. I deleted the line of doing that by mistake. You can refer to https://github.com/dragen1860/MAML-Pytorch for more information.

**Comment by 17011813** [comment]
Hello!! I ran your code with code of edited version.

But I've got the same issue with @jpainam.

How can I re-normalize weights after initialization??

Always thank you for your kind answers!

**Comment by BlueWhaleZhou** [comment]
Though the loss does not change much, the model is learning and expected to converge in a few epochs, and the evaluation is also good. One reason is that we use very limited number of labeled nodes so the model should be able to learn quickly.
